lib/openstack_query/query_executer.py

```diff
@@ -0,0 +1,113 @@
+import time
+import logging
+from typing import Callable, Optional, Dict, List, Any, Union, Type
+
+from openstack_query.runners.query_runner import QueryRunner
+from enums.cloud_domains import CloudDomains
+from enums.query.props.prop_enum import PropEnum
+
+from custom_types.openstack_query.aliases import (
+    OpenstackResourceObj,
+    ServerSideFilters,
+    ClientSideFilterFunc,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class QueryExecuter:
+    """
+    Helper class to handle executing the query - primarily performing 'run()' method
+    """
+
+    def __init__(self, prop_enum_cls: Type[PropEnum], runner: QueryRunner):
+        self._prop_enum_cls = prop_enum_cls
+        self.runner = runner
+        self._parser_func = None
+        self._output_func = None
+        self._client_side_filter_func = None
+        self._server_side_filters = None
+
+    def set_filters(
+        self,
+        client_side_filter_func: Optional[ClientSideFilterFunc] = None,
+        server_side_filters: Optional[ServerSideFilters] = None,
+    ):
+        """
+        Setter method for setting run filters
+        :param client_side_filter_func: An Optional function used to limit the results after querying openstacksdk
+        :param server_side_filters: An Optional set of filter kwargs to pass to openstacksdk
+        """
+        self._client_side_filter_func = client_side_filter_func
+        self._server_side_filters = server_side_filters
+
+    def set_parse_func(
+        self,
+        parser_func: Optional[
+            Callable[[List[OpenstackResourceObj]], Union[Dict, List]]
```

Comment:
I still really don't like this, the type basically screams to me we have a class or method here that we should be using rather than passing lambda / methods around

It's not for this PR, but we're adding more references to this so it's going to get harder to refactor :thinking: